docs/reference-guide-caffe2dml.md

```diff
@@ -0,0 +1,452 @@
+  top: "conv1"
+}
+```
+
+### Sigmoid Layer
+
+Invokes [nn/layers/tanh.dml](https://github.com/apache/systemml/blob/master/scripts/nn/layers/sigmoid.dml) layer.
+
+**Sample Usage:**
+```
+layer {
+  name: "sigmoid1"
+  type: "Sigmoid"
+  bottom: "conv1"
+  top: "conv1"
+}
+```
+
+
+### Threshold Layer
+
+Computes `X > threshold`
+
+**Parameters that are ignored:**
+- threshold (default: 0):Strictly positive values
+
+**Sample Usage:**
+```
+layer {
+  name: "threshold1"
+  type: "Threshold"
+  bottom: "conv1"
+  top: "conv1"
+}
+```
+
+## Utility Layers
+
+### Eltwise Layer
+
+Element-wise operations such as product or sum between two blobs.
+
+**Parameters that are ignored:**
+- operation(default: SUM): element-wise operation. only SUM supported for now.
+- table_prod_grad(default: true): Whether to use an asymptotically slower (for >2 inputs) but stabler method
+of computing the gradient for the PROD operation. (No effect for SUM op.)
+
+**Sample Usage:**
+```
+layer {
+	bottom: "res2a_branch1"
+	bottom: "res2a_branch2c"
+	top: "res2a"
+	name: "res2a"
+	type: "Eltwise"
+}
+```
+
+### Concat Layer
+
+**Inputs:**
+- `n_i * c_i * h * w` for each input blob i from 1 to K.
+
+**Outputs:**
+- out: Outputs, of shape
+  - if axis = 0: `(n_1 + n_2 + ... + n_K) * c_1 * h * w`, and all input `c_i` should be the same.
+  - if axis = 1: `n_1 * (c_1 + c_2 + ... + c_K) * h * w`, and all input `n_i` should be the same.
+
+**Optional Parameters:**
+- axis (default: 1): The axis along which to concatenate.
+
+**Sample Usage:**
+```
+layer {
+  name: "concat_d5cc_u5a-b"
+  type: "Concat"
+  bottom: "u5a"
+  bottom: "d5c"
+  top: "u5b"
+}
+```
+
+### Softmax Layer
+
+Computes the forward pass for a softmax classifier.  The inputs
+are interpreted as unnormalized, log-probabilities for each of
+N examples, and the softmax function transforms them to normalized
+probabilities.
+
+This can be interpreted as a generalization of the sigmoid
+function to multiple classes.
+
+`probs_ij = e^scores_ij / sum(e^scores_i)`
+
+**Parameters that are ignored:**
+- axis (default: 1): The axis along which to perform the softmax.
+
+**Sample Usage:**
+```
+layer {
+  name: "sm"
+  type: "Softmax"
+  bottom: "score"
+  top: "sm"
+}
+```
+
+## Loss Layers
+
+Loss drives learning by comparing an output to a target and assigning cost to minimize. 
+The loss itself is computed by the forward pass and the gradient w.r.t. to the loss is computed by the backward pass.
+
+### Softmax with Loss Layer
+
+The softmax loss layer computes the multinomial logistic loss of the softmax of its inputs. 
+It’s conceptually identical to a softmax layer followed by a multinomial logistic loss layer, but provides a more numerically stable gradient.
+
+Invokes [nn/layers/softmax_loss.dml](https://github.com/apache/systemml/blob/master/scripts/nn/layers/softmax_loss.dml) or
+[nn/layers/softmax2d_loss.dml](https://github.com/apache/systemml/blob/master/scripts/nn/layers/softmax2d_loss.dml) layer.
```

Comment:
minor: update these links to use a combination of `nn/layers/softmax.dml` and `nn/layers/cross_entropy_loss.dml`, or the 2D equivalents.